src/Chapter4.hs

```diff
@@ -611,6 +637,28 @@ Implement the 'Monad' instance for our lists.
   maybe a few) to flatten lists of lists to a single list.
 -}
 
+-- Cons 
+--     (Cons 1 Empty) 
+--     Cons 
+--         (Cons 2 Empty)
+--         Cons
+--             (Cons 3 Empty)
+--             Empty
+
+
+
+concatList :: List a -> List a -> List a
+concatList Empty ys = ys
+concatList (Cons x xs) ys = Cons x (concatList xs ys)
+
+flatten :: List (List a) -> List a
+flatten Empty = Empty
+flatten (Cons x xs) = concatList x (flatten xs)
+
+instance Monad List where
+    (>>=) :: List a -> (a -> List b) -> List b
+    Empty >>= _ = Empty
```

Comment:
This pattern matching is redundant. Bot `fmap` and `flatten` return `Empty` when given `Empty`, so you don't need to pattern match here separately. See, how pure functions are composable? You only need to handle corner cases in your fundamental building blocks, and you can compose them safely 🙂